runners/spark/src/main/java/org/apache/beam/runners/spark/coders/BeamSparkRunnerRegistrator.java

```diff
@@ -20,59 +20,24 @@
 
 import com.esotericsoftware.kryo.Kryo;
 import com.esotericsoftware.kryo.serializers.JavaSerializer;
-import com.google.common.base.Function;
-import com.google.common.collect.Iterables;
-import com.google.common.collect.Lists;
-import java.util.Arrays;
-import java.util.Set;
-import javax.annotation.Nullable;
-import org.apache.beam.sdk.coders.Coder;
-import org.apache.beam.sdk.io.Source;
+import org.apache.beam.runners.spark.io.MicrobatchSource;
+import org.apache.beam.runners.spark.io.SourceRDD;
 import org.apache.spark.serializer.KryoRegistrator;
-import org.reflections.Reflections;
 
 
 /**
  * Custom {@link KryoRegistrator}s for Beam's Spark runner needs.
  */
 public class BeamSparkRunnerRegistrator implements KryoRegistrator {
 
-  @Override
-  public void registerClasses(Kryo kryo) {
-    for (Class<?> clazz : ClassesForJavaSerialization.getClasses()) {
-      kryo.register(clazz, new JavaSerializer());
-    }
-  }
-
   /**
    * Register coders and sources with {@link JavaSerializer} since they aren't guaranteed to be
    * Kryo-serializable.
    */
-  private static class ClassesForJavaSerialization {
-    private static final Class<?>[] CLASSES_FOR_JAVA_SERIALIZATION = new Class<?>[]{
-        Coder.class, Source.class
-    };
-
-    private static final Iterable<Class<?>> INSTANCE;
-
-    /**
-     * Find all subclasses of ${@link CLASSES_FOR_JAVA_SERIALIZATION}
-     */
-    static {
-      final Reflections reflections = new Reflections();
-      INSTANCE = Iterables.concat(Lists.transform(Arrays.asList(CLASSES_FOR_JAVA_SERIALIZATION),
-          new Function<Class, Set<Class<?>>>() {
-            @SuppressWarnings({"unchecked", "ConstantConditions"})
-            @Nullable
-            @Override
-            public Set<Class<?>> apply(@Nullable Class clazz) {
-              return reflections.getSubTypesOf(clazz);
-            }
-          }));
-    }
-
-    static Iterable<Class<?>> getClasses() {
-      return INSTANCE;
-    }
+  @Override
+  public void registerClasses(Kryo kryo) {
+    kryo.register(MicrobatchSource.class, new JavaSerializer());
+    kryo.register(SourceRDD.Bounded.BoundedSourceWrapper.class, new JavaSerializer());
```

Comment:
I am not so familiar with kryo registration. Can you register `BoundedSource` without a wrapper, or does it not work for superclasses?